mule-migration-tool-library/src/main/java/com/mulesoft/tools/migration/library/mule/steps/salesforce/CachedBasicConfiguration.java

```diff
@@ -0,0 +1,210 @@
+
+import static com.google.common.collect.Lists.newArrayList;
+
+/**
+ * Migrate BatchJob component
+ *
+ * @author Mulesoft Inc.
+ * @since 1.0.0
+ */
+public class CachedBasicConfiguration extends AbstractApplicationModelMigrationStep implements ExpressionMigratorAware {
+
+  private static final String MULE3_NAME = "cached-basic-config";
+  private static final String MULE4_CONFIG = "sfdc-config";
+  private static final String MULE4_NAME = "basic-connection";
+  private static final String MULE4_PROXY = "proxy-configuration";
+
+  private
+
+  ExpressionMigrator expressionMigrator;
+
+  public CachedBasicConfiguration() {
+    this.setAppliedTo(XmlDslUtils.getXPathSelector(SalesforceConstants.MULE3_SALESFORCE_NAMESPACE_URI, MULE3_NAME, false));
+    this.setNamespacesContributions(newArrayList(SalesforceConstants.MULE3_SALESFORCE_NAMESPACE));
+  }
+
+  @Override
+  public void execute(Element mule3CachedBasicConfig, MigrationReport report) throws RuntimeException {
+    getApplicationModel().addNameSpace(SalesforceConstants.MULE4_SALESFORCE_NAMESPACE,
+                                       SalesforceConstants.MULE4_SALESFORCE_NAMESPACE_URI, mule3CachedBasicConfig.getDocument());
+
+    Element mule4Config = new Element(MULE4_CONFIG, SalesforceConstants.MULE4_SALESFORCE_NAMESPACE);
+    setAttributes(mule3CachedBasicConfig, mule4Config, report);
+  }
+
+  private void setAttributes(Element mule3CachedBasicConfig, Element mule4Config, MigrationReport report) {
+    //salesforce:sfdc-config
+    String nameValue = mule3CachedBasicConfig.getAttributeValue("name");
+    if (nameValue != null) {
+      mule4Config.setAttribute("name", nameValue);
+    } else {
+      report.report("salesforce.cachedBasicConfiguration", mule4Config, mule4Config);
+    }
+
+    String docName = mule3CachedBasicConfig.getAttributeValue("name", SalesforceConstants.DOC_NAMESPACE);
+    if (docName != null) {
+      mule4Config.setAttribute("name", docName, SalesforceConstants.DOC_NAMESPACE);
+    }
+
+    String fetchAllApexSoapMetadataValue = mule3CachedBasicConfig.getAttributeValue("fetchAllApexSoapMetadata");
+    if (fetchAllApexSoapMetadataValue != null) {
+      mule4Config.setAttribute("fetchAllApexSoapMetadata", fetchAllApexSoapMetadataValue);
+    }
+
+    String fetchAllApexRestMetadataValue = mule3CachedBasicConfig.getAttributeValue("fetchAllApexRestMetadata");
+    if (fetchAllApexRestMetadataValue != null) {
+      mule4Config.setAttribute("fetchAllApexRestMetadata", fetchAllApexRestMetadataValue);
+    }
+
+    //salesforce:cached-basic-connection
+    Element mule4BasicConnection = new Element(MULE4_NAME, SalesforceConstants.MULE4_SALESFORCE_NAMESPACE);
+
+    String usernameValue = mule3CachedBasicConfig.getAttributeValue("username");
+    if (usernameValue != null) {
+      mule4BasicConnection.setAttribute("username", usernameValue);
+    } else {
+      report.report("salesforce.cachedBasicConfiguration", mule4Config, mule4Config);
```

Comment:
Same as the previous comment. I think all these fields are mandatory on Mule 3 so there is no way that they will come empty.